### ISSUE TITLE
Edge case in EditableText causes infinite loop crash

### ISSUE THREAD

**Issue by rynorris** [palantir/blueprint#4267]
#### Environment

- __Package version(s)__: @blueprintjs/core 3.9.0
- __Operating System__: MacOS
- __Browser name and version__: Chrome 84.0.4147.89

#### Code Sandbox

https://codesandbox.io/s/reproduce-editabletext-bug-2nisb

#### Steps to reproduce

1. Create an `EditableText` component.
2. Set a `minWidth={17}` prop on it.
3. Type in "aa"

#### Actual behavior

The component crashes with the following error:
```
Maximum update depth exceeded. This can happen when a component repeatedly calls setState inside componentWillUpdate or componentDidUpdate. React limits the number of nested updates to prevent infinite loops.
```

![editable-text-bug](https://user-images.githubusercontent.com/3620166/89620949-6ef91600-d8cb-11ea-9d1a-441a0b87f897.gif)

#### Expected behavior

No crash.

#### Possible solution

It seems that the call to `updateInputDimensions()` [here](https://github.com/palantir/blueprint/blob/develop/packages/core/src/components/editable-text/editableText.tsx#L266) for some reason causes an infinite update loop specifically when `minWidth` is set, and entering text with very specific width.  Removing this call prevents the crash (but also prevents the component resizing of course)

**Comment by rynorris** [comment]
Upon closer inspection, it seems to be caused by the `minWidth` being set too small (probably anything smaller than the width of the text you're composing).  If I raise the `minWidth` in the example to 30, it does not crash.

I'm gonna guess the fix will be something to do with this: https://github.com/palantir/blueprint/blob/develop/packages/core/src/components/editable-text/editableText.tsx#L394

Edit:
Logging scrollWidth and minWidth just before the crash shows the loop clearly:
![image](https://user-images.githubusercontent.com/3620166/89620405-8c79b000-d8ca-11ea-8b71-05dcfd4b1644.png)

**Comment by rynorris** [comment]
Upon even closer inspection, it turns out this has nothing to do with IME composition.  I think it's just an edge case with the exact width of the characters you enter.  We were just unlucky that the Japanese characters we were entering triggered it.

As proof of that, I found a repro which doesn't require entering non-English characters.

I've updated the issue title and description with more concise descriptions and the new repro.
